### ISSUE TITLE
Can you make Sequential Data more clearly in doc?

### ISSUE THREAD

**Issue by yipliu** [PyTorchLightning/pytorch-lightning#11261]
## 📚 Documentation
Hi,

I was reading the [documentation](https://pytorch-lightning.readthedocs.io/en/stable/guides/data.html#packed-sequences-as-inputs), and it says here that Lightning has built-in support for dealing with sequential data.

Does that mean we do not need pad sequence data when we use PL?

cc @borda @rohitgr7

**Comment by rohitgr7** [comment]
padding the sequence is required to collate the sequences into a single batch and that can only happen when all the samples have same lengths. so you need to pad the sequences. It's mentioned over there: https://pytorch-lightning.readthedocs.io/en/stable/guides/data.html#packed-sequences-as-inputs

**Comment by yipliu** [comment]
> 
Sorry for my poor English.

My main puzzle was that in the original PyTorch, we also needed to pad the sequence in **DataLoader**, which is pretty routine in NLP.  However, Pytorch Lighting has an additional mention in the documentation that it has the feature ***Lightning has built-in support for dealing with sequential data***.  By built-in support, does it allow us to omit any code? Or, is it any different from PyTorch in its handling [here]( https://pytorch-lightning.readthedocs.io/en/stable/guides/data.html#packed-sequences-as-inputs)?

If not, I think the additional description [here]( https://pytorch-lightning.readthedocs.io/en/stable/guides/data.html#packed-sequences-as-inputs) would lead me to believe that PyTorch Lighting has a different approach to **Pad Sequence** that Pytorch.

 Or, is there something I don't understand clearly?

**Comment by rohitgr7** [comment]
@yipliu built-in support there doesn't mean that Lightning will handles the data stuff. More specifically it is signifying that support for sequential data like RNN sequences or iterable datasets is supported there. Padding sequences is a preprocessing step so it needs to be done by the user.

**Comment by yipliu** [comment]
Got it. Thank You!
